### PR TITLE
Feature: Gitlab webhook endpoint + LedStrip light fading

### PR DESCRIPTION
**Pull request by RickvdStaaij**
### What

So I've been working on a CIMonitor setup at home, two big features ended up in one PR, sorry for that.
1. A GitLab webhook endpoint has been added, now build and pipeline statusses can be pushed and displayed
2. The led strip module now has fading colors
### ToDo
- [x] Finish led strip module with fading lights
- [x] Replace dashboard icons
- [x] Fix travis-ci as it whines over something that works
- [x] Test on a running environment

### CODE REVIEW

**Comment by RickvdStaaij** [comment]
Not to self
```
    this.colors = {
        failure: {
            r: 255,
            g: 0,
            b: 0,
            intencity: 100
        },
        success: {
            r: 0,
            g: 255,
            b: 0,
            intencity: 100
        },
        started: {
            r: 255,
            g: 50,
            b: 0 ,
            intencity: 100
        },
        neutral: {
            r: 0,
            g: 160,
            b: 255,
            intencity: 50
        },
        off: {
            r: 0,
            g: 0,
            b: 0,
            intencity: 100
        }
    };
```

File: app/Adapter/GitLab.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    }
+
+    return ciMonitorStatus;
+};
+
+GitLab.prototype.handleBuild = function(event) {
+    var status = {
+        project: event.repository.name,
+        branch: event.ref + ' ' + event.build_name,
+        type: event.build_name.substring(0, 6) === 'deploy' ? 'deploy' : 'test',
```
Comment (RickvdStaaij) [inline app/Adapter/GitLab.js]: Try to see if there's an environment included in the body instead of the build name

Comment (RickvdStaaij) [inline app/Adapter/GitLab.js]: Doesn't work, leaving it like this for now.

**Comment by RickvdStaaij** [comment]
Finally this is done! \o/

File: app/Adapter/GitLab.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        case 'failed':
+            ciMonitorStatus = 'failure';
+            break;
+    }
+
+    return ciMonitorStatus;
+};
+
+GitLab.prototype.handleBuild = function(data) {
+    console.log('BUILD: ' + JSON.stringify(data));
```
Comment (RickvdStaaij) [inline app/Adapter/GitLab.js]: Forgot to remove :\

File: README.md [hunk 3]
```diff
@@ -43,7 +51,7 @@ limit!
     "note": "This field is optional"
 }
 ```
-1. Now for every build, the dashboard should display the status!
+2. Now for every build, the dashboard should display the status!
```
Comment (eXistenZNL) [inline README.md]: Doe gewoon overal 1. dan nummert ie zelf door.

File: public/images/types/pipeline.svg [hunk 4]
```diff
@@ -0,0 +1,42 @@
+<?xml version="1.0" encoding="iso-8859-1"?>
```
Comment (eXistenZNL) [inline public/images/types/pipeline.svg]: Ben zelf geen fan van de nieuwe icons maargoed :(

File: README.md (hunk 3 above)
Comment (RickvdStaaij) [inline README.md]: Niet bij deze, omdat er een code block tussen zit

File: public/images/types/pipeline.svg (hunk 4 above)
Comment (RickvdStaaij) [inline public/images/types/pipeline.svg]: Ik weet geen beter Icon voor een pipeline :\

Comment (eXistenZNL) [inline public/images/types/pipeline.svg]: I would opt for Financial, the right column, 4th row from the bottom. Alternatively the one on the left of that.

File: README.md (hunk 3 above)
Comment (eXistenZNL) [inline README.md]: Roger!

File: public/images/types/pipeline.svg (hunk 4 above)
Comment (RickvdStaaij) [inline public/images/types/pipeline.svg]: @eXistenZNL I think 
image
is the better alternative, will implement

Comment (RickvdStaaij) [inline public/images/types/pipeline.svg]: @eXistenZNL replaced!

Comment (eXistenZNL) [inline public/images/types/pipeline.svg]: I liked that one more, too :)

**Pull request merged**